samza-core/src/main/java/org/apache/samza/job/model/JobModelUtil.java

```diff
@@ -119,6 +120,47 @@ public static JobModel readJobModel(String jobModelVersion, MetadataStore metada
     }
   }
 
+  /**
+   * Compares the {@link ContainerModel} for a given <i>processorId</i> across two {@link JobModel}.
+   * @param processorId processor id for which work assignments are compared
+   * @param first first job model
+   * @param second second job model
+   * @return true - if {@link ContainerModel} for the processor is same across the {@link JobModel}
+   *         false - otherwise
+   */
+  public static boolean compareContainerModelForProcessor(String processorId, JobModel first, JobModel second) {
+    Preconditions.checkArgument(StringUtils.isNotBlank(processorId), "Processor id cannot be blank");
+    if (first == second) {
+      return true;
+    }
+
+    if (first == null || second == null) {
+      return false;
+    }
+
+    return compareContainerModel(first.getContainers().get(processorId), second.getContainers().get(processorId));
```

Comment:
Do you think it will be useful to add NPE guards against the result of the object obtained by `jobModel.getContainers()`

Reply:
JobModel guarantees `getContainers()` always returns non-null map instance.